lib/iris/tests/integration/test_netcdf__loadsaveattrs.py

```diff
@@ -877,120 +1020,183 @@ class TestSave(MixinAttrsTesting):
-        iris.save(cubes, self.result_filepath)
-        # Get the global+local attribute values directly from the file with netCDF4
-        if attr_name == "STASH":
-            # A special case : the stored name is different
-            attr_name = "um_stash_source"
-        try:
-            ds = threadsafe_nc4.DatasetWrapper(self.result_filepath)
-            global_result = (
-                ds.getncattr(attr_name) if attr_name in ds.ncattrs() else None
-            )
-            local_results = [
-                (
-                    var.getncattr(attr_name)
-                    if attr_name in var.ncattrs()
-                    else None
-                )
-                for var in ds.variables.values()
-            ]
-        finally:
-            ds.close()
-        return [global_result] + local_results
+        if not isinstance(values, list):
+            # Translate single input value to list-of-1
+            values = [values]
+
+        self.run_save_testcase(attr_name, [None] + values, expect_warnings)
 
-    def test_01_userstyle__single(self):
-        results = self.create_save_testcase("random", "value-x")
-        # It is stored as a *global* by default.
-        assert results == ["value-x", None]
+    def check_save_results(self, expected: list):
+        results = self.fetch_results(filepath=self.result_filepath)
+        assert results == expected
 
-    def test_02_userstyle__multiple_same(self):
-        results = self.create_save_testcase("random", "value-x", "value-x")
-        # As above.
-        assert results == ["value-x", None, None]
+    def test_userstyle__single(self, do_split):
+        self.run_save_testcase_legacytype("random", "value-x")
+        if do_split:
+            # result as input values
+            expected_result = [None, "value-x"]
+        else:
+            # in legacy mode, promoted = stored as a *global* by default.
+            expected_result = ["value-x", None]
+        self.check_save_results(expected_result)
+
+    def test_userstyle__multiple_same(self, do_split):
+        self.run_save_testcase_legacytype("random", ["value-x", "value-x"])
+        if do_split:
+            # result as input values
+            expected_result = [None, "value-x", "value-x"]
+        else:
+            # in legacy mode, promoted = stored as a *global* by default.
+            expected_result = ["value-x", None, None]
+        self.check_save_results(expected_result)
 
-    def test_03_userstyle__multiple_different(self):
-        results = self.create_save_testcase("random", "value-A", "value-B")
+    def test_userstyle__multiple_different(self):
         # Clashing values are stored as locals on the individual variables.
-        assert results == [None, "value-A", "value-B"]
+        self.run_save_testcase_legacytype("random", ["value-A", "value-B"])
+        self.check_save_results([None, "value-A", "value-B"])
+
+    def test_userstyle__multiple_onemissing(self, global_attr):
```

Comment:
This shouldn't reference `global_attr`. Possibly copy-pasted from the global equivalent of this test?